test/ProjectTemplates/Microsoft.Extensions.AI.Templates.IntegrationTests/Snapshots/aichatweb.AzureOpenAI_Qdrant_Aspire.verified/aichatweb/aichatweb.Web/Components/Pages/Chat/ChatCitation.razor

```diff
@@ -17,22 +17,23 @@
     public required string File { get; set; }
 
     [Parameter]
-    public int? PageNumber { get; set; }
-
-    [Parameter]
-    public required string Quote { get; set; }
+    public string? Quote { get; set; }
 
     private string? viewerUrl;
 
     protected override void OnParametersSet()
     {
         viewerUrl = null;
 
-        // If you ingest other types of content besides PDF files, construct a URL to an appropriate viewer here
-        if (File.EndsWith(".pdf"))
+        // If you ingest other types of content besides Markdown or PDF files, construct a URL to an appropriate viewer here
```

Comment:
Suggestion, but don't block this PR on it: We could refactor all the places where we need to switch on the document type and have a `DocumentTypeHandling` utility class that pulls all of this together. It would have methods for getting the mime type by extension, getting the viewer url from a citation, and doing the reading. That would centralize all of the things that need to be extended to support additional file formats.